day_two_guide/topics/managing_docker_registries.adoc

```diff
@@ -0,0 +1,659 @@
+
+. Use the `dockercfg` secret to pull images from the authenticated registry by
+linking the secret to the service account performing the pull operations. The
+default service account to pull images is named `default`:
++
+----
+$ oc secrets link default <my_registry> --for=pull
+----
+
+. For pushing images using the S2I feature, the `dockercfg` secret is mounted
+in the S2I pod, so it needs to be linked to the proper service account that
+performs the build. The default service account used to build images is named
+`builder`.
++
+----
+$ oc secrets link builder <my_registry>
+----
+
+. In the `buildconfig`, the secret should be specified for push or pull
+operations:
++
+----
+"type": "Source",
+"sourceStrategy": {
+    "from": {
+        "kind": "DockerImage",
+        "name": "*my.registry.example.com*/myproject/myimage:stable"
+    },
+    "pullSecret": {
+        "name": "*mydockerregistry*"
+    },
+...[OUTPUT ABBREVIATED]...
+"output": {
+    "to": {
+        "kind": "DockerImage",
+        "name": "*my.registry.example.com*/myproject/myimage:latest"
+    },
+    "pushSecret": {
+        "name": "*mydockerregistry*"
+    },
+...[OUTPUT ABBREVIATED]...
+----
+
+. If the external registry delegates authentication to external services, create
+both `dockercfg` secrets: the registry one using the registry URL and the
+external authentication system using its own URL. Both secrets should be added
+to the service accounts as well.
++
+----
+$ oc project <my_project>
+$ oc secrets new-dockercfg <my_registry> --docker-server=*<my_registry_example.com> --docker-username=<username> --docker-password=<my_password> --docker-email=<me@example.com>
+$ oc secrets new-dockercfg <my_docker_registry_ext_auth> --docker-server=<my.authsystem.example.com> --docker-username=<username> --docker-password=<my_password> --docker-email=<me@example.com>
+$ oc secrets link default <my_registry> --for=pull
+$ oc secrets link default <my_docker_registry_ext_auth> --for=pull
+$ oc secrets link builder <my_registry>
+$ oc secrets link builder <my_docker_registry_ext_auth>
+----
+
+==== ImagePolicy admission plug-in
+
+An admission control plug-in intercepts requests to the API, and performs checks
+depending on the rules it has configured and allows or denies certain actions
+based on those rules. {product-title} can limit the allowed images running in
+the environment
+xref:../admin_guide/image_policy.adoc#admin-guide-image-policy[using the
+`ImagePolicy` admission plug-in] where it can control:
+
+* The source of images: which registries can be used to pull images
+* Image resolution: force pods to run with immutable digests to ensure the
+image does not change due to a re-tag
+* Container image label restrictions: force an image to have or not have
+particular labels
+* Image annotation restrictions: force an image in the integrated container
+registry to have or not have particular annotations
+
+WARNING: ImagePolicy admission plug-in is currently considered beta.
```

Comment:
Is Beta the same as tech preview? If not, is there a better way to explain the supportability here?

Reply:
[Here](https://github.com/openshift/openshift-docs/blob/master/admin_guide/image_policy.adoc#overview) says beta as well. I'm not sure on who to ask about this, @cooktheryan any clue?

Reply:
May have to ping Marc C as he is always in the know of what's ga and tech preview